### PR TITLE
build(js): Add "jest-staged" yarn test to run tests related to staged files only

### PR DESCRIPTION
**Pull request by billyvg**
`yarn jest-staged` == yarn jest --findRelatedTests $(git diff --cached --name-only)

### CODE REVIEW

**Review by MaxBittker — APPROVED** [review]
this is a good improvement in itself, but i do think our proliferation of test* and jest* scripts has gotten confusing. WTB json comments here.

File: package.json [hunk 1]
```diff
@@ -93,10 +93,9 @@
   "proxyURL": "http://localhost:8000",
   "scripts": {
     "test": "node scripts/test.js",
-    "test-ci": "npm run test -- --runInBand --ci --coverage --testResultsProcessor=jest-junit",
-    "test-watch": "npm run test -- --watch",
```
Comment (billyvg) [inline package.json]: redundant since `watch` is now default

File: package.json [hunk 2]
```diff
@@ -93,10 +93,9 @@
...
-    "jest": "node scripts/test.js",
```
Comment (billyvg) [inline package.json]: removed so that we can run default jest via `yarn jest` if needed

File: package.json [hunk 3]
```diff
@@ -93,10 +93,9 @@
...
-    "jest-debug": "node --inspect-brk scripts/test.js --runInBand",
```
Comment (billyvg) [inline package.json]: renamed to `test-debug`

**Review by billyvg — COMMENTED** [review]
@MaxBittker cleaned up a few things

**Comment by MaxBittker** [comment]
these look great. 👍

**Pull request merged**